### PR TITLE
Add ClockDividerAndMux util module

### PR DESCRIPTION
**Pull request by jerryz123**

### CODE REVIEW

**Review by abejgonzalez — APPROVED** [review]
LGTM

**Review by colinschmidt — APPROVED** [review]
LGTM, with just one question.

File: src/main/scala/ClockUtil.scala [hunk 1]
```diff
@@ -164,6 +164,26 @@ class ClockDivider(width: Int) extends Module {
 
 }
 
+class ClockDivideAndMux(width: Int, depth: Int = 3, genClockGate: () => ClockGate) extends Module {
```
Comment (abejgonzalez) [inline src/main/scala/ClockUtil.scala:167]: Do you care about stopping the clock? Why not just make this be the divider on `divisor = [1, N)` and the passthrough on `divisor = 0`? No subtract operation needed?

Comment (jerryz123) [reply src/main/scala/ClockUtil.scala:167]: Good point. I actually prefer this way. If we want to stop the clock, we can use the existing clock gating stuff.

File: src/main/scala/ClockUtil.scala [hunk 2]
```diff
@@ -84,11 +84,11 @@ class ClockMutexMux(val n: Int, depth: Int, genClockGate: () => ClockGate) exten
     val io = IO(new Bundle {
         val clocksIn = Input(Vec(n, Clock()))
         val clockOut = Output(Clock())
-        val resetAsync = Input(Bool())
+        val resetAsync = Input(Reset())
```
[the remark this answers is not in the document]
Comment (jerryz123) [reply src/main/scala/ClockUtil.scala]: Fixed

**Review by abejgonzalez — APPROVED** [review]
LGTM. I approved the other version anyways... I figured you wanted to keep the "clock stop" functionality.

**Comment by jerryz123** [comment]
Clock stop functionality can always be added with a clock gate on the output clock of this block, which I think is a neater way to do this.

**Pull request merged**